rapid_response_xblock/static/js/src/rapid.js

```diff
@@ -133,28 +120,129 @@
 
     /**
      * Renders the graph and adjusts axes based on responses to the given problem.
+     */
+    function renderD3() {
+      // Get the indexes for selected runs. This should either be [0] or [0, 1].
+      var chartKeys = _.keys(state.selectedRuns);
+
+      // D3 data join for charts. Create a container div for each chart to store graph, select element and buttons.
+      var containers = d3.select(element).select(
+        rapidBlockResultsSel
+      ).selectAll(".chart-container").data(chartKeys);
+      var containersEnter = containers.enter()
+        .append("div");
+
+      // chart selection, close and compare buttons
+      var selectionRows = containers.selectAll(".selection-row");
+      var selectionRowsEnter = containersEnter
+        .append("div")
+        .classed("selection-row", true);
```

Comment:
It's a little counterintuitive but this is basically what's happening:
 - `selectionRows` is set to be a selector covering all **currently existing** `.selection-row` elements
 - `selectionRowsEnter` is set to be a selector of **new** elements, corresponding to new items in the data. These elements need to be created at this point and all initialization happens here. The [D3 data join example](https://bl.ocks.org/mbostock/3808218) is more elegant but we need a separate `selectionRowsEnter` variable here because we need to append multiple elements to the root, not just one.
 - At this point there is no overlap between `selectionRows` and `selectionRowsEnter`. To cover both new elements and already existing elements `selectionRowsMerged` is created using the `.merge()` function. `selectionRowsMerged` now applies to all elements, new and old, so it's a good place to set attributes and properties.
 - At the bottom of the function there is `containers.exit().remove()`. We could have a `selectionRows.exit().remove()` to remove elements where the data item was removed, but it will already be taken care of here since `containers` is a parent selector for that element.